src/features/blacksmith/Blacksmith.tsx

```diff
@@ -35,23 +35,28 @@ export const Blacksmith: React.FC = () => {
         top: `${GRID_WIDTH_PX * 6}px`,
       }}
     >
-      <img
-        src={blacksmith}
-        alt="market"
-        onClick={isNotReadOnly ? () => open() : undefined}
-        className={classNames("w-full", {
+      <div
+        className={classNames({
           "cursor-pointer": isNotReadOnly,
           "hover:img-highlight": isNotReadOnly,
         })}
-      />
-      {isNotReadOnly && (
-        <Action
-          className="absolute -bottom-8 left-1"
-          text="Craft"
-          icon={hammer}
-          onClick={open}
+      >
+        <img
+          src={blacksmith}
+          alt="market"
+          onClick={isNotReadOnly ? openBlacksmith : undefined}
```

Comment:
Similar to `openBank` situation

Reply:
Moved to openBlacksmith function. Thanks! 😄